src/dataflow/components/dataflow-program.tsx

```diff
@@ -220,6 +223,36 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   private setProgramRunTime = (time: number) => {
     this.setState({programRunTime: time});
   }
+  private generateProgramData = () => {
+    let programName: any = "my-program";
+    const endTimePad = 24 * 60 * 60 * 1000;
+    const newTimestamp = Date.now() + endTimePad;
+    const hubs: string[] = [];
+    const sensors: string[] = [];
+    this.programEditor.nodes.forEach((n: Node) => {
+      if (n.name === "Sensor" && n.data.sensor) {
+        const chInfo = this.channels.find(ci => ci.channelId === n.data.sensor);
+        if (chInfo) {
+          hubs.push(chInfo.hubId);
+          sensors.push(`${chInfo.hubId}_${chInfo.channelId}`);
+        }
+      } else if (n.name === "Data Storage" && n.data.datasetName) {
+        programName = n.data.datasetName;
+      }
+    });
+    const programData = {
+      program: {
+        endTime: newTimestamp,
+        hubs,
+        // tslint:disable-next-line:max-line-length
+        program: "\"{\"file_version\":\"0.1\",\"blocks\":[{\"type\":\"sensor\",\"source\":\"2711-tempe\",\"save\":true,\"label\":\"My temperature\"},{\"type\":\"sensor\",\"source\":\"2711-humid\",\"save\":true,\"label\":\"My humidity\"}]}\"",
```

Comment:
Can we line break this in a useful way to eliminate turning off the linter?

Reply:
This won't be in here, so no need to disable the linter. - this comes from the programEditor JSON (or some transformed JSON based off that if we decided not to use it directly).

Reply:
@dstrawberrygirl Or, for now, can I just hand you the Rete JSON?

Reply:
Great.  I'll add that in.

Reply:
Fixed.  I am now setting the rete JSON in the `program` field.